rest-server/admin/src/main/scala/com/actionml/admin/MongoAdministrator.scala

```diff
@@ -77,62 +101,43 @@ class MongoAdministrator extends Administrator with JsonParser with Mongo {
   Success/failure indicated in the HTTP return code
   Action: creates or modifies an existing engine
   */
-  def addEngine(json: String): Validated[ValidateError, EngineId] = {
+  def addEngine(json: String)(implicit ec: ExecutionContext): Future[Validated[ValidateError, EngineId]] = {
     // val params = parse(json).extract[GenericEngineParams]
```

Comment:
Is the ExecutionContext where we can use Scaldi to inject from? I'd like the inject the Mongo implementation for the DAOs so the Engine do not need to deal with Mongo specific details. Same for the Mirror implementation. These should be set with configuration and injected at runtime.

Reply:
don't bother with injection now, just asking.